test/systemtests/kubeadm_test.go

```diff
@@ -0,0 +1,981 @@
+package systemtests
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/contiv/contivmodel/client"
+	"os"
+	"time"
+)
+
+type kubePod struct {
+	node *node
+}
+
+var k8sMaster *node
+
+const (
+	netmasterRestartFile = "/tmp/restart_netmaster"
+	netpluginRestartFile = "/tmp/restart_netplugin"
+	netmasterLogLocation = "/var/contiv/log/netmaster.log"
+	netpluginLogLocation = "/var/contiv/log/netplugin.log"
+)
+
+//var master sync.Mutex
+
+func (s *systemtestSuite) NewK8sPodExec(n *node) *kubePod {
+	k8 := new(kubePod)
+	k8.node = n
+
+	if n.Name() == "k8master" {
+		k8sMaster = n
+	}
+	return k8
+}
+
+func (s *systemtestSuite) TearDownInfraNetwork() error {
+	err := s.cli.NetworkDelete("default", "default-net")
+	if err != nil {
+		logrus.Errorf("default-net not deleted. Err: %+v", err)
+		return err
+	}
+	time.Sleep(time.Second)
+	return nil
+}
+
+func (s *systemtestSuite) SetupInfraNetwork() error {
+	err := s.cli.NetworkPost(&client.Network{
+		TenantName:  "default",
+		NetworkName: "default-net",
+		Subnet:      "100.10.1.0/24",
+		Encap:       "vxlan",
+	})
+	if err != nil {
+		logrus.Errorf("default-net not created. Err: %+v", err)
+		return err
+	}
+	time.Sleep(time.Second)
+	return nil
+}
+
+func (k *kubePod) newContainer(node *node, containerID, name string, spec containerSpec) (*container, error) {
```

Comment:
can't change the name that easily. It's tied up at the framework level. Will change if this function is just used locally within this file.